Source/Core/DolphinQt/QtUtils/InputSuppressorFocusFilter.cpp

```diff
@@ -0,0 +1,78 @@
+// Copyright 2025 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "DolphinQt/QtUtils/InputSuppressorFocusFilter.h"
+
+#include <QAbstractItemView>
+#include <QAbstractSpinBox>
+#include <QComboBox>
+#include <QEvent>
+#include <QFocusEvent>
+#include <QGuiApplication>
+#include <QLineEdit>
+#include <QObject>
+#include <QTextEdit>
+
+#include "Core/InputSuppressor.h"
+
+namespace
+{
+struct InputSuppressorFocusOutFilter : public QObject
+{
+public:
+  InputSuppressorFocusOutFilter(QObject* const object) : QObject(object) {}
+
+  bool eventFilter(QObject* const object, QEvent* const event) override
+  {
+    if (object == parent() && event->type() == QEvent::FocusOut)
+    {
+      object->removeEventFilter(this);
+      InputSuppressor::RemoveSuppression();
```

Comment:
I have a fear (and maybe it's unjustified) that an object could be deleted without getting a `FocusOut` event and leave input suppression stuck.

I guess it's probably safe to rely on a perfect one-to-one relationship of `FocusIn` and `FocusOut` events?

But I think I'd sleep better if `InputSuppressorFocusOutFilter` just held an `InputSuppressor` instance to automatically add/remove suppression on construction/destruction.